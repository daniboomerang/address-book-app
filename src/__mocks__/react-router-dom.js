import React from 'react';

const rrd = require('react-router-dom');

// Just render a fragment with its children
rrd.BrowserRouter = ({ children }) => <>{children}</>;
module.exports = rrd;
