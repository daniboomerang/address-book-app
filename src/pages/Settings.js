import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faArrowLeft } from '@fortawesome/free-solid-svg-icons';
import { Link } from 'react-router-dom';
import NationalityOptionsGrid from '../components/NationalityOptionsGrid';

/**
 * Adreess book settings page
 * It renders the different nationality options for users fetching
 */
export const Settings = () => (
  <div className="w-full h-full">
    <div className="w-full flex">
      <Link to="/">
        <FontAwesomeIcon
          className="m-4 transform cursor-pointer duration-200 hover:scale-110 hover:text-red-600 z-20"
          icon={faArrowLeft}
          fixedWidth={false}
          size="2x"
        />
      </Link>
    </div>
    <NationalityOptionsGrid />
  </div>
);
export default Settings;
