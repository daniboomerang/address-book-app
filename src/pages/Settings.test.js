import React from 'react';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import Settings from './Settings';

describe('Settings component', () => {
  it('should correctly render the settings page', () => {
    const wrapper = mount(
      <MemoryRouter keyLength={0}>
        <Settings />
      </MemoryRouter>
    );

    expect(wrapper.find('FontAwesomeIcon').length).toBe(1);
    expect(wrapper.find('NationalityOptionsGrid').length).toBe(1);

    expect(wrapper).toMatchSnapshot();
  });
});
