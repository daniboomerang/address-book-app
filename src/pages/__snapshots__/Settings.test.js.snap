// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings component should correctly render the settings page 1`] = `
<MemoryRouter
  keyLength={0}
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Settings
      selectedNationalities={Array []}
    >
      <div
        className="w-full h-full"
      >
        <div
          className="w-full flex"
        >
          <Link
            to="/"
          >
            <LinkAnchor
              href="/"
              navigate={[Function]}
            >
              <a
                href="/"
                onClick={[Function]}
              >
                <FontAwesomeIcon
                  border={false}
                  className="m-4 transform cursor-pointer duration-200 hover:scale-110 hover:text-red-600 z-20"
                  fixedWidth={false}
                  flip={null}
                  icon={
                    Object {
                      "icon": Array [
                        448,
                        512,
                        Array [],
                        "f060",
                        "M257.5 445.1l-22.2 22.2c-9.4 9.4-24.6 9.4-33.9 0L7 273c-9.4-9.4-9.4-24.6 0-33.9L201.4 44.7c9.4-9.4 24.6-9.4 33.9 0l22.2 22.2c9.5 9.5 9.3 25-.4 34.3L136.6 216H424c13.3 0 24 10.7 24 24v32c0 13.3-10.7 24-24 24H136.6l120.5 114.8c9.8 9.3 10 24.8.4 34.3z",
                      ],
                      "iconName": "arrow-left",
                      "prefix": "fas",
                    }
                  }
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size="2x"
                  spin={false}
                  swapOpacity={false}
                  symbol={false}
                  title=""
                  transform={null}
                >
                  <svg
                    aria-hidden="true"
                    className="svg-inline--fa fa-arrow-left fa-w-14 fa-2x m-4 transform cursor-pointer duration-200 hover:scale-110 hover:text-red-600 z-20"
                    data-icon="arrow-left"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    style={Object {}}
                    viewBox="0 0 448 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M257.5 445.1l-22.2 22.2c-9.4 9.4-24.6 9.4-33.9 0L7 273c-9.4-9.4-9.4-24.6 0-33.9L201.4 44.7c9.4-9.4 24.6-9.4 33.9 0l22.2 22.2c9.5 9.5 9.3 25-.4 34.3L136.6 216H424c13.3 0 24 10.7 24 24v32c0 13.3-10.7 24-24 24H136.6l120.5 114.8c9.8 9.3 10 24.8.4 34.3z"
                      fill="currentColor"
                      style={Object {}}
                    />
                  </svg>
                </FontAwesomeIcon>
              </a>
            </LinkAnchor>
          </Link>
        </div>
        <NationalityOptionsGrid
          selectedNationalities={Array []}
        >
          <div
            className="px-6 max-w-md m-auto"
          >
            <div
              className="h-20 my-6 text-center"
              data-testid="options-grid-message"
            >
              Filter your users by nationalies.
              <div
                className="text-sm"
              >
                If no nationalities are selected no filter will be applied
              </div>
            </div>
            <div
              className="grid grid-cols-2 justify-between items-center m-auto col-gap-4 row-gap-12"
              style={
                Object {
                  "justifyItems": "center",
                }
              }
            >
              <NationalityOptionCard
                isSelected={false}
                key="CH"
                nationality="CH"
              >
                <button
                  className="focus:outline-none grid border-4 text-center justify-center transform cursor-pointer duration-200 hover:border-red-600 hover:text-red-600 pt-4 pb-2 md:pt-5 md:pb-3 w-24 sm:w-32 hover:rounded-md border-blue-500"
                  data-testid="nationality-CH"
                  onClick={[Function]}
                  type="button"
                >
                  <ChFlag
                    className="h-12 w-12 sm:h-24 sm:w-24"
                  >
                    <svg
                      className="h-12 w-12 sm:h-24 sm:w-24"
                      data-testid="CH-flag"
                      viewBox="0 0 512 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <rect
                        fill="#D80027"
                        height="512"
                        width="512"
                      />
                      <polygon
                        fill="#F0F0F0"
                        points="395.13,209.624 302.377,209.624 302.377,116.87 209.623,116.87 209.623,209.624   116.87,209.624 116.87,302.377 209.623,302.377 209.623,395.13 302.377,395.13 302.377,302.377 395.13,302.377 "
                      />
                    </svg>
                  </ChFlag>
                  <span
                    className="mt-4"
                  >
                    CH
                  </span>
                </button>
              </NationalityOptionCard>
              <NationalityOptionCard
                isSelected={false}
                key="ES"
                nationality="ES"
              >
                <button
                  className="focus:outline-none grid border-4 text-center justify-center transform cursor-pointer duration-200 hover:border-red-600 hover:text-red-600 pt-4 pb-2 md:pt-5 md:pb-3 w-24 sm:w-32 hover:rounded-md border-blue-500"
                  data-testid="nationality-ES"
                  onClick={[Function]}
                  type="button"
                >
                  <EsFlag
                    className="h-12 w-12 sm:h-24 sm:w-24"
                  >
                    <svg
                      className="h-12 w-12 sm:h-24 sm:w-24"
                      data-testid="ES-flag"
                      viewBox="0 0 512 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <rect
                        fill="#FFDA44"
                        height="512"
                        width="512"
                      />
                      <g>
                        <rect
                          fill="#D80027"
                          height="170.663"
                          width="512"
                        />
                        <rect
                          fill="#D80027"
                          height="170.663"
                          width="512"
                          y="341.337"
                        />
                      </g>
                    </svg>
                  </EsFlag>
                  <span
                    className="mt-4"
                  >
                    ES
                  </span>
                </button>
              </NationalityOptionCard>
              <NationalityOptionCard
                isSelected={false}
                key="BR"
                nationality="BR"
              >
                <button
                  className="focus:outline-none grid border-4 text-center justify-center transform cursor-pointer duration-200 hover:border-red-600 hover:text-red-600 pt-4 pb-2 md:pt-5 md:pb-3 w-24 sm:w-32 hover:rounded-md border-blue-500"
                  data-testid="nationality-BR"
                  onClick={[Function]}
                  type="button"
                >
                  <BrFlag
                    className="h-12 w-12 sm:h-24 sm:w-24"
                  >
                    <svg
                      className="h-12 w-12 sm:h-24 sm:w-24"
                      data-testid="BR-flag"
                      viewBox="0 0 512 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <rect
                        fill="#F0F0F0"
                        height="512"
                        width="512"
                      />
                      <polygon
                        fill="#D80027"
                        points="0,304 208,304 208,512 304,512 304,304 512,304 512,208 304,208 304,0 208,0 208,208 0,208   "
                      />
                      <g>
                        <polygon
                          fill="#0052B4"
                          points="406.92,333.913 512,438.993 512,333.913  "
                        />
                        <polygon
                          fill="#0052B4"
                          points="333.913,333.913 512,512 512,461.64 384.273,333.913  "
                        />
                        <polygon
                          fill="#0052B4"
                          points="464.564,512 333.913,381.336 333.913,512  "
                        />
                      </g>
                      <polygon
                        fill="#F0F0F0"
                        points="333.913,333.913 512,512 512,461.64 384.273,333.913 "
                      />
                      <polygon
                        fill="#D80027"
                        points="333.913,333.913 512,512 512,461.64 384.273,333.913 "
                      />
                      <g>
                        <polygon
                          fill="#0052B4"
                          points="80.302,333.913 0,414.215 0,333.913  "
                        />
                        <polygon
                          fill="#0052B4"
                          points="178.084,356.559 178.084,511.997 22.658,511.997  "
                        />
                      </g>
                      <polygon
                        fill="#D80027"
                        points="127.724,333.916 0,461.641 0,512 0,512 178.084,333.916 "
                      />
                      <g>
                        <polygon
                          fill="#0052B4"
                          points="105.08,178.087 0,73.007 0,178.087  "
                        />
                        <polygon
                          fill="#0052B4"
                          points="178.087,178.087 0,0 0,50.36 127.727,178.087  "
                        />
                        <polygon
                          fill="#0052B4"
                          points="47.436,0 178.087,130.663 178.087,0  "
                        />
                      </g>
                      <polygon
                        fill="#F0F0F0"
                        points="178.087,178.087 0,0 0,50.36 127.727,178.087 "
                      />
                      <polygon
                        fill="#D80027"
                        points="178.087,178.087 0,0 0,50.36 127.727,178.087 "
                      />
                      <g>
                        <polygon
                          fill="#0052B4"
                          points="431.698,178.087 512,97.785 512,178.087  "
                        />
                        <polygon
                          fill="#0052B4"
                          points="333.916,155.441 333.916,0.003 489.342,0.003  "
                        />
                      </g>
                      <polygon
                        fill="#D80027"
                        points="384.276,178.084 512,50.359 512,0 512,0 333.916,178.084 "
                      />
                    </svg>
                  </BrFlag>
                  <span
                    className="mt-4"
                  >
                    BR
                  </span>
                </button>
              </NationalityOptionCard>
              <NationalityOptionCard
                isSelected={false}
                key="FR"
                nationality="FR"
              >
                <button
                  className="focus:outline-none grid border-4 text-center justify-center transform cursor-pointer duration-200 hover:border-red-600 hover:text-red-600 pt-4 pb-2 md:pt-5 md:pb-3 w-24 sm:w-32 hover:rounded-md border-blue-500"
                  data-testid="nationality-FR"
                  onClick={[Function]}
                  type="button"
                >
                  <FrFlag
                    className="h-12 w-12 sm:h-24 sm:w-24"
                  >
                    <svg
                      className="h-12 w-12 sm:h-24 sm:w-24"
                      data-testid="FR-flag"
                      viewBox="0 0 512 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <rect
                        fill="#F0F0F0"
                        height="512"
                        width="512"
                      />
                      <rect
                        fill="#0052B4"
                        height="512"
                        width="170.663"
                      />
                      <rect
                        fill="#D80027"
                        height="512"
                        width="170.663"
                        x="341.337"
                      />
                    </svg>
                  </FrFlag>
                  <span
                    className="mt-4"
                  >
                    FR
                  </span>
                </button>
              </NationalityOptionCard>
            </div>
          </div>
        </NationalityOptionsGrid>
      </div>
    </Settings>
  </Router>
</MemoryRouter>
`;
